[cutlass backend] Global filter ops before situation based filter ops

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #157866

The idea of this PR is that, sometimes we are filtering ops based not based on the node specific information. For example, we always filter out simt ops. So I want to group them together into a global filtering function. 

This can help shrink the config space as well. 20s -> 6s for instantiation 3332.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend] Global filter ops before situation based filter ops



Differential Revision: [D77949178](https://our.internmc.facebook.com/intern/diff/D77949178/)

[ghstack-poisoned]

ColinPeppler commented on torch/_inductor/codegen/cuda/cutlass_utils.py:
```diff
@@ -348,6 +353,10 @@ def get_accumulator_dtype(
     Given a pair of input torch dtypes, returns the inferred accumulator torch dtype.
     """
 
+    assert OrderedSet(input_torch_dtypes) <= XW_DTYPES, (
```
`input_torch_dtypes in XW_DTYPE` doesn't work?

ColinPeppler commented on torch/_inductor/codegen/cuda/gemm_template.py:
```diff
@@ -800,6 +814,51 @@ def _dtype_match(
+            != cutlass_lib.OpcodeClass.Simt
+        ]
+
+        # only keep the set of row x column ops
```
so we ignore RR, CR, CC? a comment explaining why would help a lot.

henrylhtsang committed:
Update on "[cutlass backend] Global filter ops before situation based filter ops"


Differential Revision: [D77949178](https://our.internmc.facebook.com/intern/diff/D77949178/)

The idea of this PR is that, sometimes we are filtering ops based not based on the node specific information. For example, we always filter out simt ops. So I want to group them together into a global filtering function. 

This can help shrink the config space as well. 20s -> 6s for instantiation 3332.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot rebase -b viable/strict

henrylhtsang committed:
Update on "[cutlass backend] Global filter ops before situation based filter ops"


The idea of this PR is that, sometimes we are filtering ops based not based on the node specific information. For example, we always filter out simt ops. So I want to group them together into a global filtering function. 

This can help shrink the config space as well. 20s -> 6s for instantiation 3332.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge
